mlxtend/frequent_patterns/fpcommon.py

```diff
@@ -48,25 +48,48 @@ def generate_itemsets(generator, num_itemsets, colname_map):
 
 
 def valid_input_check(df):
-    # Fast path: if all columns are boolean, there is nothing to check
-    if not (df.dtypes == bool).all():
-        # Pandas is much slower than numpy, so use df.values instead of df here
-        idxs = np.where((df.values != 1) & (df.values != 0))
-        if len(idxs[0]) > 0:
-            val = df.values[idxs[0][0], idxs[1][0]]
-            s = ('The allowed values for a DataFrame'
-                 ' are True, False, 0, 1. Found value %s' % (val))
-            raise ValueError(s)
-
-    is_sparse = hasattr(df, "to_coo")
-    if is_sparse:
+    if hasattr(df, "to_coo"):
         if not isinstance(df.columns[0], str) and df.columns[0] != 0:
             raise ValueError('Due to current limitations in Pandas, '
                              'if the SparseDataFrame has integer column names,'
                              'names, please make sure they either start '
                              'with `0` or cast them as string column names: '
                              '`df.columns = [str(i) for i in df.columns`].')
 
+    # Fast path: if all columns are boolean, there is nothing to check
+    all_bools = False
+    if hasattr(df, "sparse"):
+        try:
+            # SparseDtype introduced in pandas 0.24; works for DataFrame
+            # with SparseArray, or with deprecated SparseDataFrame
+            all_bools = (df.dtypes == pd.SparseDtype(bool)).all()
+        except AttributeError:
```

Comment:
Regarding the try-except and `AttributeError` Is this an issue that would be raised in pandas < 0.24? In that case, we could also use 

```python
    from distutils.version import LooseVersion as Version
    from pandas import __version__ as pandas_version

    if Version(pandas_version) >= Version("0.24"):
        all_bools = (df.dtypes == pd.SparseDtype(bool)).all()
```

or am I missing something?

Reply:
It's a matter of taster, I find attribute checking more robust than version checking, but this is your code, so I changed it as requested ;-)